fix(greynoise): Connector crashes with KeyError: 'bot' in internal-enrichment

## Description

The GreyNoise internal-enrichment connector consistently fails to process IPv4 enrichment results with a KeyError: 'bot'. The connector successfully calls the GreyNoise v3 API and receives a response, but crashes while building labels because internet_scanner_intelligence.bot is not present in the response for some IPs. This makes enrichment fail for any IP where that field is absent, with the error reported back to OpenCTI instead of a completed enrichment.

This seems to be leftover regression from the SDK v2 to v3 migration in #6028, which changed the code from flat data["bot"] to nested data["internet_scanner_intelligence"]["bot"] and, in the same commit, removed bot from the documented label list in README.md but didn't update the code to guard against the field being missing.

## Environment

OS (where OpenCTI server runs): Red Hat Enterprise Linux release 9.7
OpenCTI version: 7.260515.0
OpenCTI client: Connector (opencti/connector-greynoise:7.260515.0)
Other environment details:
  - GreyNoise API: v3
  - Valid API key with proper authentication

## Reproducible steps

1. Deploy the GreyNoise internal-enrichment connector with a valid API key
2. Trigger enrichment on an IPv4 observable whose GreyNoise internet_scanner_intelligence response does not include a bot field (e.g. 46.101.101.175)
3. Connector reads the StixCyberObservable, calls the GreyNoise API, and begins generating the STIX bundle
4. _process_labels accesses data["internet_scanner_intelligence"]["bot"] directly and raises KeyError

## Expected output

INFO: [CONNECTOR] IPv4, has been identified by GreyNoise and generation of the Stix bundle is in progress.
INFO: [CONNECTOR] For this Ipv4, the number of Stix bundle(s) that will be enriched...

## Actual output

The connector fails with a KeyError instead of completing enrichment.

Error logs:

{"timestamp": "2026-06-24T14:50:31.637246Z", "level": "INFO", "name": "GreyNoise", "message": "[CONNECTOR] IPv4, has been identified by GreyNoise and generation of the Stix bundle is in progress.",
"attributes": {"IPv4": "46.101.101.175"}}
{"timestamp": "2026-06-24T14:50:31.678868Z", "level": "ERROR", "name": "GreyNoise", "message": "Error in message processing, reporting error to API", "exc_info": "Traceback (most recent call last):\n  File
\"/opt/opencti-connector-greynoise/main.py\", line 908, in _process_message\n    stix_bundle(json_data, stix_entity)\n^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/opencti-connector-greynoise/main.py\", line 832, in _generate_stix_bundle\n    labels, malwares = self._process_labels(data)\n
     ^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/opencti-connector-greynoise/main.py\", lin  if data[\"internet_scanner_intelligence\"][\"bot\"] is True:\n~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^\nKeyError: 'bot'\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File
\"/usr/local/lib/python3.12/site-packages/pycti/connector/opencti_connector_helper.py\",    message = self.callback(event_data)\n              ^^^^^^^^^^^^^^^^^^^^^^^^^\n  File\"/opt/opencti-connector-greynoise/main.py\", line 939, in _process_message\n    raise ValueError(\nValueError: ('[ERROR] Unexpected Error occurred:', {'Exception': \"'bot'\"})"}

## Additional information

The README change implies bot is no longer a reliable field in the v3 internet_scanner_intelligence object, but main.py:220 still does a direct, unguarded dict access instead of .get("bot"), the same defensive pattern already used a few lines above it for tags:

entity_tags = data["internet_scanner_intelligence"].get("tags", [])

Proposed Solution:
internal-enrichment/greynoise/src/main.py:220

# Before
if data["internet_scanner_intelligence"]["bot"] is True:

# After
if data["internet_scanner_intelligence"].get("bot") is True:

Impact:
- Current: enrichment fails and reports an error for any IP whose GreyNoise v3 response omits bot
- After fix: enrichment completes normally for those IPs, simply the "Known BOT Activity" label is absent

--- a/internal-enrichment/greynoise/src/main.py
+++ b/internal-enrichment/greynoise/src/main.py
@@ -217,7 +217,7 @@ class GreyNoiseConnector:
                 "#57B9FF",
             )
 
-        if data["internet_scanner_intelligence"]["bot"] is True:
+        if data["internet_scanner_intelligence"].get("bot") is True:
             # Create label for "Known Bot Activity"
             self._create_custom_label("Known BOT Activity", "#7e4ec2")
 

